### DISCUSSION TITLE
Which API provides insertable snippets throughout the editor like this?

### DISCUSSION THREAD

**Discussion by tinder-tannerbennett** [microsoft/vscode-discussions#643] [Extension Development QnA] [answered]
I can't seem to find anything about it

<img width="450" alt="Screenshot 2023-06-13 at 4 54 23 PM" src="https://github.com/microsoft/vscode-discussions/assets/89099176/d7fbf8f8-327d-476f-8280-9841b7378f72">

**Comment by r3m0t** [comment, marked as answer]
`InlayHint.textEdits`

**Reply by tinder-tannerbennett** [reply]
Thank you! Do inlay hints provide a way to disable the "double click to insert" feature? Swift's language extension provides hints like these in the screenshot, and while helpful, they are not valid code and should not ever be inserted

i.e. can you provide an empty array to `textEdits`? Or will it still default to inserting the hint at that point?
